src/EventProducerService.cpp

```diff
@@ -33,21 +33,22 @@ void EventProducerService::setProducedEvents()
 { 
   for (byte i = 1; i <= module_config->EE_PRODUCED_EVENTS; i++)
   {
-    createDefaultEvent(i);    
+    createDefaultEvent(i);
+    opCode = OPC_ACOF;
+    generateMessage();
```

Comment:
I don't think we should generate messages here as no input has changed yet. No existing CBUS modules do this. 
And are you sure the input state is OFF? :)

Reply:
The message is arbitrary.  I used off so as not to leave anything on!  If you don't do this, then you won't see the defaults in the FCU until you read events in the nodes menu.  It was an after-thought in doing this but as valid as sending a message when restoring a default.  I don't consider what existing CBUS modules may or may not do as being a good guide.  It can come out easily enough if you feel strongly about it.

Reply:
I still don't like these generated events.
1) As you say the off state is arbitrary, and may indicate the wrong state. Shouldn't matter as these events won't be taught to any other nodes at this stage. But it just doesn't feel right.
2) Say the user doesn't want a few of these events and changes their NVs. If these events are created before the user gets a chance to change these NVs, then those extra events will show up in FCU anyway. And we know how messy it is to delete events.
3) As you have seen before, I do use existing CBUS modules as inspiration. What works well for them should be kept. If we want to do something different we have to consider any other side effects.

Reply:
1. It seems logical that a state should be initialised off.
2. They're produced upon initialisation and so are there anyway even if they can't yet be seen in the FCU.  However, as soon a you read the events, it will appear in the FCU even if the NV is 0.  Now, we could not create any defaults until the event tries to be produced, a bit like re-instating a default previously deleted.  However, this could get tedious if there are a lot.  On the other hand, maybe if there are a lot as in Ian's Universal, it should have a dedicated service.
3. I also take inspiration from existing CBUS modules but that should not constrain us.